File: database/src/main/java/org/difft/app/database/models/PendingMessageModelNew.java
package org.difft.app.database.models;

import com.tencent.wcdb.WCDBField;
import com.tencent.wcdb.WCDBTableCoding;

import java.util.Arrays;
import java.util.Objects;

@WCDBTableCoding
public class PendingMessageModelNew {

    @WCDBField(isPrimary = true, isUnique = true)
    public String messageId;

    @WCDBField
    public long originalMessageTimeStamp;

    @WCDBField(isNotNull = true)
    public byte[] messageEnvelopBytes = new byte[0];

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        PendingMessageModelNew that = (PendingMessageModelNew) o;
        return Objects.equals(messageId, that.messageId) && Objects.equals(originalMessageTimeStamp, that.originalMessageTimeStamp) && Objects.deepEquals(messageEnvelopBytes, that.messageEnvelopBytes);
    }

    @Override
    public int hashCode() {
        return Objects.hash(messageId, originalMessageTimeStamp, Arrays.hashCode(messageEnvelopBytes));
    }
}

File: database/src/main/java/org/difft/app/database/models/PendingMessageModelNew.kt
package org.difft.app.database.models

import com.tencent.wcdb.WCDBField
import com.tencent.wcdb.WCDBTableCoding
import java.util.Arrays
import java.util.Objects

@WCDBTableCoding
class PendingMessageModelNew {

    @WCDBField(isPrimary = true, isUnique = true)
    var messageId: String? = null

    @WCDBField
    var originalMessageTimeStamp: Long = 0

    @WCDBField(isNotNull = true)
    var messageEnvelopBytes: ByteArray = ByteArray(0)

    override fun equals(other: Any?): Boolean {
        if (other == null || javaClass != other.javaClass) return false
        other as PendingMessageModelNew
        return messageId == other.messageId &&
                originalMessageTimeStamp == other.originalMessageTimeStamp &&
                Objects.deepEquals(messageEnvelopBytes, other.messageEnvelopBytes)
    }

    override fun hashCode(): Int =
        Objects.hash(messageId, originalMessageTimeStamp, Arrays.hashCode(messageEnvelopBytes))
}
